algorithms.cpp

```diff
@@ -131,6 +132,24 @@ namespace sift {
             return bins;
         }
 
+        const std::vector<f32_t> orientationHistogram8(
+                const vigra::MultiArray<2, f32_t>& orientations,
+                const vigra::MultiArray<2, f32_t>& magnitudes, 
+                const vigra::MultiArray<2, f32_t>& current_gauss) {
+
+            std::vector<f32_t> bins(8, 0);
+            for (u16_t x = 0; x < orientations.width(); x++) {
+                for (u16_t y = 0; y < orientations.height(); y++) {
+                    const f32_t sum = magnitudes(x, y) * current_gauss(x, y);
+                    u16_t i = std::floor(orientations(x, y) / 45);
+                    i = i > 7 ? 0 : i;
+                    bins[i] += sum;
+                }
+            }
+            return bins;
```

Comment:
also in der methode mach ich nen array befüll das um am ende alles in nen vector zu kopieren und den zu returnen?